catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VisualPlacementBrickTest.java

```diff
@@ -51,62 +56,129 @@
+		Script script =
+				BrickTestUtils.createProjectAndGetStartScript(VisualPlacementBrickTest.class.getSimpleName());
+		script.addBrick(brick);
+
 		baseActivityTestRule.launchActivity();
 	}
 
-	@Category({Cat.AppUi.class, Level.Smoke.class})
 	@Test
-	public void testPlaceAtBrick() {
-		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_place_at);
-		onView(withId(R.id.brick_place_at_edit_text_x))
+	public void testIsVisualPlacementShown() {
+		onBrickAtPosition(1).checkShowsText(brickString);
+		onView(withId(editTextXId))
 				.perform(click());
 		onView(withText(R.string.brick_place_at_option_place_visually))
 				.check(matches(isDisplayed()));
 		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
-				.check(matches(isDisplayed()))
+				.check(matches(isDisplayed()));
+		pressBack();
+
+		onView(withId(editTextYId))
+				.perform(click());
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
+				.check(matches(isDisplayed()));
+		pressBack();
+	}
+
+	@Test
+	public void testIsVisualPlacementShownOnBrickClick() {
+		onBrickAtPosition(1).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()));
+		pressBack();
+	}
+
+	@Test
+	public void testNoVisualPlacementAfterFormulaEdited() {
+		onView(withId(editTextXId))
+				.perform(click());
+		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
 				.perform(click());
 		onFormulaEditor()
-				.performEnterFormula("1+2")
+				.performEnterFormula(firstInput)
 				.performCloseAndSave();
-		onView(withId(R.id.brick_place_at_edit_text_x))
-				.check(matches(withText("1 + 2 ")));
-
-		onView(withId(R.id.brick_place_at_edit_text_x))
+		onView(withId(editTextXId))
+				.check(matches(withText(expectedFirstOutput)))
 				.perform(click());
 		onFormulaEditor()
-				.performEnterNumber(42)
+				.performEnterNumber(0)
 				.performCloseAndSave();
-		onView(withId(R.id.brick_place_at_edit_text_x))
-				.check(matches(withText("42 ")));
+	}
 
-		onView(withId(R.id.brick_place_at_edit_text_y))
+	@Test
+	public void testVisualPlacementAfterNumberEntered() {
+		onView(withId(editTextXId))
+				.perform(click());
+		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
+				.perform(click());
+		onFormulaEditor()
+				.performEnterNumber(secondInput)
+				.performCloseAndSave();
+		onView(withId(editTextXId))
+				.check(matches(withText(expectedSecondOutput)))
 				.perform(click());
 		onView(withText(R.string.brick_place_at_option_place_visually))
 				.check(matches(isDisplayed()));
 		onView(withText(R.string.brick_context_dialog_formula_edit_brick))
 				.check(matches(isDisplayed()));
 		pressBack();
-
-		onBrickAtPosition(brickPosition).performClick();
-		onView(withText(R.string.brick_place_at_option_place_visually))
-				.check(matches(isDisplayed()));
 	}
```

Comment:
Please add a test case that verifies that the visual placement activity is shown after the click (or a similar test ;)). Have a look at Mockito :D.